pandas/core/tools/datetimes.py

```diff
@@ -103,6 +106,70 @@ def _convert_and_box_cache(arg, cache_array, box, errors, name=None):
     return result.values
 
 
+def _return_parsed_timezone_results(result, tznames, tzoffsets, parsing_tzname,
+                                    parsing_tzoffset, box):
+    """
+    Return results from array_strptime if a %z or %Z directive was passed.
+
+    If %Z is only parsed, timezones will be a pytz.timezone object.
+    If %z is only parsed, timezones will be a pytz.FixedOffset object.
+    If both %Z and %z are parsed, timezones will be a datetime.timezone object.
+
+    Parameters
+    ----------
+    result : ndarray
+        int64 date representations of the dates
+    tznames : ndarray
+        strings of timezone names if %Z is parsed
+    tzoffsets : ndarray
+        timedelta objects of the timezone offset if %z is parsed
+    parsing_tzname : boolean
+        True if %Z is parsed
+    parsing_tzoffset : boolean
+        True if %z is parsed
+    box : boolean
+        True boxes result as an Index-like, False returns an ndarray
+
+    Returns
+    -------
+    tz_result : ndarray of parsed dates with timezone
+        Returns:
+
+        - Index-like if box=True
+        - ndarray of Timestamps if box=False
+
+    """
+    if parsing_tzname and not parsing_tzoffset:
+        tz_results = np.array([tslib.Timestamp(res, tz=tz)
+                               for res, tz in zip(result, tznames)])
+    elif parsing_tzoffset and not parsing_tzname:
+        tz_results = []
+        for res, offset in zip(result, tzoffsets):
+            offset_mins = offset.total_seconds() / 60
+            tzoffset = pytz.FixedOffset(offset_mins)
+            ts = tslib.Timestamp(res)
+            ts = ts.tz_localize(tzoffset)
+            tz_results.append(ts)
+        tz_results = np.array(tz_results)
```

Comment:
what do you need all of this for, this is jumping thru a lot of hoops here

Reply:
This `elif` branch is:

1) Creating a `pytz.FixedOffset` from the parsed offset
2) Creating a naive Timestamp, then localizing it to the `pytz.FixedOffset` (can't do it directly like `Timezone(res, tz=pytz.FixedOffset(...))` because of my realization from #20257)